### PR TITLE
[stable/airflow] Remove connection before adding it

### LINKED ISSUE #23499: [stable/airflow] When upgrading the existing airflow chart, scheduler.connections changes do not take effect
**Describe the bug**

When the stable/airflow chart is first installed, the internal add-connections.sh script adds connections to airflow defined by the `scheduler.connections:` section of the values.yaml file. When the chart is subsequently upgraded with changes to the `scheduler.connections:` section, the changes do not take effect. I would have expected that the current connection values were reflected in airflow.

**Version of Helm and Kubernetes**:

Helm v3.2.4
K8S 1.17

**Which chart**:

stable/airflow

**How to reproduce it** (as minimally and precisely as possible):

Install stable/airflow with one set of values in `scheduler.connections:` 

Upgrade the existing stable/airflow chart with a different set of values in `scheduler.connections:`

### PR DESCRIPTION
**Pull request by vsimon**
Addresses #23499
The --delete command exits 0 (success) even if the connection does not
exist.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #23499

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by thesuperzapper** [comment]
Thanks for the contribution!

I would say this qualifies as a bug, so should be a minor version bump. 
My only hesitation is that people may have modified a connection using the Admin UI, and added a password, and this would remove those changes.

Could you rebase against master (which would make the `Chart.yaml` version: `7.4.1`

**Comment by vsimon** [comment]
Sure, I'll rebase. That's a good hesitation, perhaps putting this around a conditional i.e. `scheduler.refreshConnections` bool  (default false) under the Airflow Scheduler values in values.yaml might help with that?

**Comment by vsimon** [comment]
Pushed a draft of the conditional added.

File: stable/airflow/values.yaml [hunk 1]
```diff
@@ -236,6 +236,10 @@ scheduler:
   ##
   connections: []
 
+  ## if we remove before adding a connection resulting in a refresh
+  ##
+  refreshConnections: false
```
Comment (thesuperzapper) [inline stable/airflow/values.yaml]: I think the expected default behaviour is probably `true`.

Also, as this is a top-level value, please add a description in the readme.md values table.

File: stable/airflow/README.md [hunk 2]
```diff
@@ -107,6 +107,9 @@ scheduler:
         }
 ```
 
+We expose the `scheduler.refreshConnections` value to refresh connections by
+removing them before adding them when they are automatically being imported.
+
```
Comment (thesuperzapper) [inline stable/airflow/README.md:114]: If we are making `scheduler.refreshConnections` default to `true`, perhaps mention that if a user wants to add a password after the initial deployment, they should set `scheduler.refreshConnections` to `false`

File: stable/airflow/Chart.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Airflow is a platform to programmatically author, schedule and monitor workflows
 name: airflow
-version: 7.4.0
+version: 7.5.0
```
Comment (thesuperzapper) [inline stable/airflow/Chart.yaml]: For each minor version bump, we should add to `UPGRADING.md` and update the link in `README.md`.

However, this could be considered a bug-fix, so I think bumping to `7.4.1` is probably acceptable.

**Comment by thesuperzapper** [comment]
Also can you remove the random space diffs, which cause issues with other pending PRs.

File: stable/airflow/templates/secret-connections.yaml [hunk 4]
```diff
@@ -1,6 +1,10 @@
 {{- define "connections.script" }}
   #!/usr/bin/env bash
+
```
Comment (thesuperzapper) [inline stable/airflow/templates/secret-connections.yaml]: why is this here?

Comment (vsimon) [reply stable/airflow/templates/secret-connections.yaml]: accidentally introduced, will remove.

File: stable/airflow/Chart.yaml (hunk 3 above)
Comment (vsimon) [reply stable/airflow/Chart.yaml]: OK sounds good then for 7.4.1.

File: stable/airflow/README.md (hunk 2 above)
Comment (vsimon) [reply stable/airflow/README.md:114]: added the mention

File: stable/airflow/values.yaml (hunk 1 above)
Comment (vsimon) [reply stable/airflow/values.yaml]: added to the readme.md values table

**Comment by vsimon** [comment]
> Also can you remove the random space diffs, which cause issues with other pending PRs.

Yeah. I had to find how to turn off trim trailing whitespace in the editor. I could split this out into a commit or open in a new PR?

**Comment by thesuperzapper** [comment]
@vsimon feel free to put it in a seperate commit

**Comment by thesuperzapper** [comment]
@vsimon, can you prepend the commit message with `[stable/airflow]`

**Comment by retrry** [comment]
> Thanks for the contribution!
> 
> I would say this qualifies as a bug, so should be a minor version bump.
> My only hesitation is that people may have modified a connection using the Admin UI, and added a password, and this would remove those changes.
> 
> Could you rebase against master (which would make the `Chart.yaml` version: `7.4.1`

So your hesitation was true. This "minor" version bump broke my setup :(

**Comment by thesuperzapper** [comment]
@retrry sorry :(
